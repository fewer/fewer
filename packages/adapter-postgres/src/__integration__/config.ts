export default {
  database: 'postgres',
  user: 'postgres',
  password: 'password',
  host: 'postgres',
};
