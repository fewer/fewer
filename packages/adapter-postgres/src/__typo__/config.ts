export default {
  database: 'fewer_integration_tests',
  user: 'postgres',
  password: 'password',
  host: 'postgres',
};
