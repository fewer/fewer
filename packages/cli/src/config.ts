import cosmiconfig from 'cosmiconfig';

export interface FewerConfigurationFile {
  src: string;
  migrations: string;
  repositories: string;
  databases: string[];
  typescript: boolean;
  cjs: boolean;
}

const DEFAULT_CONFIG: FewerConfigurationFile = {
  src: 'src',
  migrations: 'src/migrations',
  repositories: 'src/repositories',
  databases: ['src/database.ts'],
  typescript: true,
  cjs: false,
};

let userConfig: FewerConfigurationFile | undefined;
const explorer = cosmiconfig('fewer');
const result = explorer.searchSync(process.cwd());

if (result) {
  // TODO: Should we validate the format here on runtime?
  userConfig = result.config as FewerConfigurationFile;
}

const config: FewerConfigurationFile = {
  ...DEFAULT_CONFIG,
  ...userConfig,
};

export default config;
