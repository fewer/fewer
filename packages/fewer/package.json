{
  "name": "fewer",
  "version": "0.0.6",
  "description": "A minimal ORM for Node.js.",
  "keywords": [
    "orm",
    "sql",
    "fewer"
  ],
  "author": "Jordan Gensler <jgensler@netflix.com>",
  "homepage": "https://github.com/fewer/fewer",
  "license": "MIT",
  "main": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/fewer/fewer.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/fewer/fewer/issues"
  },
  "dependencies": {
    "@fewer/sq": "^0.0.4"
  },
  "devDependencies": {
    "@fewer/typeval": "^0.0.4"
  }
}
