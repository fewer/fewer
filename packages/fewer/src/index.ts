import Schema from './Schema';
import { Repository, createRepository } from './Repository';

export { Schema, Repository, createRepository };
