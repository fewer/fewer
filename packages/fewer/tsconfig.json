{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib",
    "composite": true,
    "rootDir": "src"
  },
  "include": ["./src"],
  "references": [
    { "path": "../sq" },
  ]
}
