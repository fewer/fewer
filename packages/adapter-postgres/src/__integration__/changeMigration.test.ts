import { createMigration, createDatabase, Database } from 'fewer';
import { Adapter } from '..';
import config from './config';

describe('change migration', () => {
  describe('postgres', () => {
    let database: Database<Adapter>;
    let adapter: Adapter;

    beforeAll(async () => {
      adapter = new Adapter(config);

      database = createDatabase({ adapter });

      await database.connect();
    });

    afterAll(async () => {
      await database.disconnect();
    });

    it('can run createTable up and down', async () => {
      const migration = createMigration(database, {
        change: (m, t) =>
          m.createTable('users', null, {
            id: t.bigserial({ primaryKey: true }),
            firstName: t.string(),
            lastName: t.string(),
            unbounded: t.varchar(),
            email: t.string({ nonNull: true }),
          }),
      });

      migration.run('up');

      await adapter.migrate(migration);

      expect(
        await adapter.rawQuery(
          "select table_name, column_name, column_default, is_nullable, data_type, character_maximum_length, is_generated, is_updatable from INFORMATION_SCHEMA.COLUMNS where table_name = 'users';",
        ),
      ).toMatchSnapshot();
    });
  });
});
