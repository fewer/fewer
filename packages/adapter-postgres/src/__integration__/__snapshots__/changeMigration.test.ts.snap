// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change migration postgres can run createTable up and down 1`] = `
Array [
  Object {
    "character_maximum_length": null,
    "column_default": "nextval('users_id_seq'::regclass)",
    "column_name": "id",
    "data_type": "bigint",
    "is_generated": "NEVER",
    "is_nullable": "NO",
    "is_updatable": "YES",
    "table_name": "users",
  },
  Object {
    "character_maximum_length": 255,
    "column_default": null,
    "column_name": "firstname",
    "data_type": "character varying",
    "is_generated": "NEVER",
    "is_nullable": "YES",
    "is_updatable": "YES",
    "table_name": "users",
  },
  Object {
    "character_maximum_length": 255,
    "column_default": null,
    "column_name": "lastname",
    "data_type": "character varying",
    "is_generated": "NEVER",
    "is_nullable": "YES",
    "is_updatable": "YES",
    "table_name": "users",
  },
  Object {
    "character_maximum_length": null,
    "column_default": null,
    "column_name": "unbounded",
    "data_type": "character varying",
    "is_generated": "NEVER",
    "is_nullable": "YES",
    "is_updatable": "YES",
    "table_name": "users",
  },
  Object {
    "character_maximum_length": 255,
    "column_default": null,
    "column_name": "email",
    "data_type": "character varying",
    "is_generated": "NEVER",
    "is_nullable": "YES",
    "is_updatable": "YES",
    "table_name": "users",
  },
]
`;
